libs/common/src/state-migrations/migration-helper.spec.ts

```diff
@@ -0,0 +1,84 @@
+import { MockProxy, mock } from "jest-mock-extended";
+
+// eslint-disable-next-line import/no-restricted-paths -- Needed to print log messages
+import { LogService } from "../abstractions/log.service";
+// eslint-disable-next-line import/no-restricted-paths -- Needed to interface with storage locations
+import { AbstractStorageService } from "../abstractions/storage.service";
+
+import { MigrationHelper } from "./migration-helper";
+
+const exampleJSON = {
+  authenticatedAccounts: [
+    "c493ed01-4e08-4e88-abc7-332f380ca760",
+    "23e61a5f-2ece-4f5e-b499-f0bc489482a9",
+  ],
+  "c493ed01-4e08-4e88-abc7-332f380ca760": {
+    otherStuff: "otherStuff",
+  },
+  "23e61a5f-2ece-4f5e-b499-f0bc489482a9": {
+    otherStuff: "otherStuff",
+  },
```

Comment:
Suggestion only: I recommend disambiguating the contents of these accounts, this way the test will throw if the sut happens to mix them up:

```suggestion
  "c493ed01-4e08-4e88-abc7-332f380ca760": {
    otherStuff: "otherStuff1",
  },
  "23e61a5f-2ece-4f5e-b499-f0bc489482a9": {
    otherStuff: "otherStuff2",
  },
```

Same for the individual migration specs where applicable.